tools/jit/gen_jit_dispatch.py

```diff
@@ -472,7 +472,11 @@ def format_arg(arg):
         decl = '{} {}'.format(typ, name)
         if 'default' in arg:
             # clean up initializer lists {{true, true}} -> [true, true]
-            default = str(arg['default']) \
+            default = arg['default']
+            # NOTE: str(float) in python2 truncates, which makes JIT signatures not match native_functions
+            # signatures.  repr(float) doesn't seem to truncate in these cases.
+            default = str(default) if not isinstance(default, float) else repr(default)
```

Comment:
Is there any reason we can't use repr universally here?

Reply:
yes, it's not guaranteed to preserve the representation of other types.  For example, it adds quotes around a string.